File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/activity/SplashActivity.java
package com.andruid.magic.newsdaily.activity;

import android.content.Intent;

import com.andruid.magic.newsdaily.R;
import com.andruid.magic.newsdaily.util.PrefUtil;
import com.daimajia.androidanimations.library.Techniques;
import com.viksaa.sssplash.lib.activity.AwesomeSplash;
import com.viksaa.sssplash.lib.cnst.Flags;
import com.viksaa.sssplash.lib.model.ConfigSplash;

import java.util.concurrent.Executors;

public class SplashActivity extends AwesomeSplash {
    @Override
    public void initSplash(ConfigSplash configSplash) {
        configSplash.setBackgroundColor(R.color.colorSplash);
        configSplash.setAnimCircularRevealDuration(1500);
        configSplash.setRevealFlagX(Flags.REVEAL_RIGHT);
        configSplash.setRevealFlagY(Flags.REVEAL_BOTTOM);

        configSplash.setLogoSplash(R.mipmap.ic_launcher_foreground);
        configSplash.setAnimLogoSplashDuration(1500);
        configSplash.setAnimLogoSplashTechnique(Techniques.Bounce);

        configSplash.setTitleSplash(getString(R.string.app_name));
        configSplash.setTitleTextColor(R.color.colorPrimary);
        configSplash.setTitleTextSize(30f);
        configSplash.setAnimTitleDuration(1500);
        configSplash.setAnimTitleTechnique(Techniques.FlipInX);
    }

    @Override
    public void animationsFinished() {
        Executors.newSingleThreadExecutor().execute(() -> {
            if(PrefUtil.isFirstTime(SplashActivity.this)){
                runOnUiThread(() -> {
                    startActivity(new Intent(SplashActivity.this, IntroActivity.class));
                    finish();
                });
                PrefUtil.updateFirstTimePref(this);
            }
            else{
                runOnUiThread(() -> {
                    startActivity(new Intent(SplashActivity.this, MainActivity.class));
                    finish();
                });
            }
        });
    }
}

File: NewsDailyJava/app/src/main/java/com/andruid/magic/newsdaily/activity/SplashActivity.kt
package com.andruid.magic.newsdaily.activity

import android.content.Intent
import com.andruid.magic.newsdaily.R
import com.andruid.magic.newsdaily.util.PrefUtil.Companion.isFirstTime
import com.andruid.magic.newsdaily.util.PrefUtil.Companion.updateFirstTimePref
import com.daimajia.androidanimations.library.Techniques
import com.viksaa.sssplash.lib.activity.AwesomeSplash
import com.viksaa.sssplash.lib.cnst.Flags
import com.viksaa.sssplash.lib.model.ConfigSplash

class SplashActivity : AwesomeSplash() {

    override fun initSplash(configSplash: ConfigSplash) {
        configSplash.apply {
            backgroundColor = R.color.colorSplash
            animCircularRevealDuration = 1500
            revealFlagX = Flags.REVEAL_RIGHT
            revealFlagY = Flags.REVEAL_BOTTOM

            logoSplash = R.mipmap.ic_launcher_foreground
            animLogoSplashDuration = 1500
            animLogoSplashTechnique = Techniques.Bounce

            titleSplash = getString(R.string.app_name)
            titleTextColor = R.color.colorPrimary
            titleTextSize = 30f
            animTitleDuration = 1500
            animTitleTechnique = Techniques.FlipInX
        }
    }

    override fun animationsFinished() {
        if (isFirstTime(this@SplashActivity)) {
            startActivity(Intent(this@SplashActivity, IntroActivity::class.java))
            finish()
            updateFirstTimePref(this)
        } else {
            startActivity(Intent(this@SplashActivity, MainActivity::class.java))
            finish()
        }
    }
}